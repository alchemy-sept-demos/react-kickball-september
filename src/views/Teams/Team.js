import React from 'react';
import { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import { Link } from 'react-router-dom';
import { fetchTeamById } from '../../services/teams';

export default function Team(props) {
  const [team, setTeam] = useState({ players: [] });
  // VERSION 2 -- uses useParams hook instead of props
  const params = useParams();
  console.log(params);

  useEffect(() => {
    const fetchData = async () => {
      // VERSION 1
      const data = await fetchTeamById(props.match.params.id);
      // VERSION 2
      // const data = await fetchTeamById(params.id);
      setTeam(data[0]);
    };
    fetchData();
  }, []);
  return (
    <div>
      <h1>{team.name}</h1>
      <h2>Players</h2>
      {team.players.map((player) => (
        <Link to={`/players/${player.id}`} key={player.id}>
          {player.name}
        </Link>
      ))}
    </div>
  );
}
