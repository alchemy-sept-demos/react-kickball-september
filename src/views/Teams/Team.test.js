import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import Team from './Team';

test("should render a team's details", async () => {
  // render the component

  // VERSION 1
  // When the route uses the component prop
  // <Route path="/teams/:id" component={Team} />
  // The component itself is getting the id information from the props
  // we can fake that out in our test by passing props that looks like the
  // react router props
  // Since our component uses a <Link> we do need to wrap our test in a
  // MemoryRouter but since its not using anything else from React Router
  // we don't have to redefine routes
  const { container } = render(
    <MemoryRouter>
      <Team match={{ params: { id: 2 } }} />
    </MemoryRouter>
  );

  // VERSION 2
  // When the route does not use component prop (uses useParams hook instead)
  // i.e. looks like this and the component gets the id from useParams
  // <Route path="/teams/:id">
  //    <Team />
  // </Route>
  // Because the Team component is dependent on React Router hook, we have to
  // render it inside of MemoryRouter AND generate the Route inside of the test
  // so that the test has all the information it needs to render
  // const { container } = render(
  //   <MemoryRouter initialEntries={['/teams/2']}>
  //     <Route path="/teams/:id">
  //       <Team />
  //     </Route>
  //   </MemoryRouter>
  // );

  await screen.findByText('Stumptown Lumberjacks');
  expect(container).toMatchSnapshot();
});
