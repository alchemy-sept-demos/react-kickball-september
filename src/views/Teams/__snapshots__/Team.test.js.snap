// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a team's details 1`] = `
<div>
  <div>
    <h1>
      Stumptown Lumberjacks
    </h1>
    <h2>
      Players
    </h2>
    <a
      href="/players/5"
    >
      Gary Coolman
    </a>
    <a
      href="/players/6"
    >
      Terry Cruise
    </a>
    <a
      href="/players/7"
    >
      Bud E. Guy
    </a>
    <a
      href="/players/8"
    >
      Buster Rimes
    </a>
  </div>
</div>
`;
